### PR TITLE
CON-1185: PoC Python enclave API using Jep

### PR DESCRIPTION
**Pull request by shamsasari**
Using Jep to create a wrapper around the existing Java enclave API in Python. This has been achieved by implementing a normal Kotlin Enclave class, `PythonEnclaveAdapter`, which delegates the various calls into to Python code which is provided by the user as a script file. This “adapter” enclave is implemented in a new python-enclave-adapter module. Currently only mock mode works and it must be created manually via the `createMockHost` internal method.

The user will consume the Python API via duck typing. `PythonEnclaveAdapter` will look to see if the matching global function is defined and if so execute it:

* `on_enclave_startup()` - executed by `onStartup`
* `on_enclave_shutdown()` - executed by `onShutdown`
* `receive_from_untrusted_host(bytes)` - executed by `receiveFromUntrustedHost`. The Java byte array is converted to Python `bytes`. If there’s no return value then it is treated as null, otherwise the return value is expected to be `bytes`.
* `receive_enclave_mail(mail)` - executed by `receiveMail`. The Java `EnclaveMail` object is converted to a simpler Python version which is just a class holding the body, envelope and authenticated sender. The topic and sequence number are ignored for now. The authenticated sender is represented by its encoded binary form in `bytes`. This function is also much simpler than `receiveMail` in that the response can simply be returned from the function and it will encrypted as a response mail back to the sender key and posted.

There’s also a global `enclave_sign(data)` function available which delegates to the Conclave enclave’s signing ability via `signer()`.

### CODE REVIEW

**Review by bonifazi-r3 — CHANGES_REQUESTED** [review]
Everything looks good and interesting here.
Got another request: do we need to add  a `gradle/wrapper` things here?
Also, a minimal README?
Finally, I am getting a `prepareKotlinBuildScriptModel` related error, which I found can be fixed by adding
```
task prepareKotlinBuildScriptModel {

}
```
in the main `build.gradle` file. Is that normal?

File: python-enclave-adapter/src/test/kotlin/com/r3/conclave/python/PythonEnclaveAdapterTest.kt [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        assertThat(pythonValue<Boolean>("on_startup_called")).isEqualTo(true)
+    }
+
+    @Test
+    fun `on_enclave_startup() is optional`() {
+        loadEnclave("""
+            def on_enclave_shutdown():
+                pass
+        """.trimIndent())
+    }
```
[the remark this answers is not in the document]
Comment (shamsasari) [reply python-enclave-adapter/src/test/kotlin/com/r3/conclave/python/PythonEnclaveAdapterTest.kt:66]: I'll add a comment.

**Comment by shamsasari** [comment]
README added, with instructions on how to run the tests.

**Review by bonifazi-r3 — APPROVED** [review]
Thanks for this.

**Pull request merged**